frontend/src/app/lib/InstallManager.js

```diff
@@ -198,10 +198,16 @@ export function setupAddonConnection(store) {
 
 export function enableExperiment(dispatch, experiment, sendToGA) {
   if (!mam) {
-    return;
+    return Promise.reject(new Error("no mozAddonManager"));
   }
 
-  mam
+  dispatch(
+    updateExperiment(experiment.addon_id, {
+      inProgress: true
+    })
+  );
+
+  return mam
```

Comment:
The previous PR that changed this file stopped returning a promise. I didn't catch it then so I'm fixing it now.